### PR TITLE
[pm] Cleaning  unused includes in Tree library

### PR DESCRIPTION
**Pull request by oshadura**
…y for Tree library

### CODE REVIEW

File: tree/tree/src/TTree.cxx [hunk 1]
```diff
@@ -385,7 +385,6 @@ End_Macro
 #include "TTreeCacheUnzip.h"
 #include "TVirtualCollectionProxy.h"
 #include "TEmulatedCollectionProxy.h"
-#include "TVirtualFitter.h"
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx]: Is this redundant (then why is it) or unnecessary (not used)?

**Comment by oshadura** [comment]
@pcanal Tree is using a TMath.h in many places in source code - e.g. https://github.com/root-project/root/blob/master/tree/tree/src/TBasket.cxx#L19 [TMath::Max()]
How it didn't crash before, it is interesting question. Probably because MathCore was used in many other places, triggering to be build before building Tree, for example in Core.

**Comment by pcanal** [comment]
The few use of TMath in TTree are through inline function (Min, Max, Sort, BinarySearch).

**Comment by oshadura** [comment]
@pcanal, sorry for bad example, in case of  Min() and Max() it is true that it is inlined functions that are actually part of Core, but it is not true for Sort(), BinarySearch(). I will have this PR only as a removal of not used includes and I will address your comments about MathCore in different PR, since I think I have a better solution for Sort, BinarySearch functions :)

**Comment by pcanal** [comment]
> but it is not true for Sort(), BinarySearch(). 

They are inline functon in TMath.h which is part of mathcore.  So your change is fine but the facts they are inlined also explain why the library is actually not needed.

**Pull request merged**